modules/carbonAnalyticsAdapter.js

```diff
@@ -0,0 +1,227 @@
+import { generateUUID } from '../src/utils.js';
+import { ajax } from '../src/ajax.js';
+import { getStorageManager } from '../src/storageManager.js';
+import {getGlobal} from '../src/prebidGlobal.js';
+import adapterManager from '../src/adapterManager.js';
+import adapter from '../libraries/analyticsAdapter/AnalyticsAdapter.js';
+import CONSTANTS from '../src/constants.json';
+
+const CARBON_GVL_ID = 493;
+const ANALYTICS_VERSION = 'v1.0';
+const CARBON_ANALYTICS_URL = 'https://pb-ing.ccgateway.net';
+const PROFILE_ID_KEY = 'carbon_ccuid';
+const PROFILE_ID_COOKIE = 'ccuid';
+const SESSION_ID_COOKIE = 'ccsid';
+const MODULE_NAME = 'carbon';
+const ANALYTICS_TYPE = 'endpoint';
+
+const DAY_MS = 24 * 60 * 60 * 1000;
+const MINUTE_MS = 60 * 1000;
+
+export const storage = getStorageManager({gvlid: CARBON_GVL_ID, moduleName: 'carbon'});
+
+let enabledEvents = [];
+let pageViewId = '';
+let profileId = '';
+let sessionId = '';
+let parentId = '';
+let pageEngagement = {};
+
+let carbonAdapter = Object.assign(adapter({CARBON_ANALYTICS_URL, ANALYTICS_TYPE}), {
+  track({eventType, args}) {
+    if (!enabledEvents.includes(eventType)) return;
+    args = args ? JSON.parse(JSON.stringify(args)) : {};
+    switch (eventType) {
+      case CONSTANTS.EVENTS.AUCTION_END: {
+        registerEngagement();
+        let event = createBaseEngagementEvent(args);
+        sendEngagementEvent(event, 'auction_end');
+        break;
+      }
+      case CONSTANTS.EVENTS.TCF2_ENFORCEMENT: {
+        registerEngagement();
+        let event = createBaseEngagementEvent(args);
+
+        event.tcf_events = args; // Do we only want relevant information here?
+
+        sendEngagementEvent(event, 'tcf_enforcement');
+        break;
+      }
+    }
+  }
+});
+
+// save the base class function
+carbonAdapter.originEnableAnalytics = carbonAdapter.enableAnalytics;
+
+// override enableAnalytics so we can get access to the config passed in from the page
+carbonAdapter.enableAnalytics = function (config) {
+  if (config.options) {
+    parentId = config.options.parentId;
+    if (config.options.enabledEvents) {
+      enabledEvents = config.options.enabledEvents;
+    }
+  }
+
+  pageViewId = generateUUID();
+  profileId = getProfileId();
+  sessionId = getSessionId();
+
+  pageEngagement = { // create the initial page engagement event
+    ttl: 60 * 1, // should be multiple of 60, unit is seconds
+    count: 0,
+    id: generateUUID(),
+    timeLastEngage: Date.now()
+  };
+
+  let event = createBaseEngagementEvent()
+  sendEngagementEvent(event, 'page_load');
+
+  carbonAdapter.originEnableAnalytics(config); // call the base class function
+};
+
+function getProfileId() {
+  if (storage.localStorageIsEnabled) {
```

Comment:
`localStorageIsEnabled` is a function you have to execute to get the actual result. This would always be truthy.

`storage.localStorageIsEnabled()`